dot-net-sdk/EppoClient.cs

```diff
@@ -178,6 +233,173 @@ public static EppoClient Init(EppoClientConfig eppoClientConfig)
+    ///     }
+    /// };
+    /// var result = client.GetBanditAction(
+    ///     "flagKey",
+    ///     "subjecKey",
+    ///     subjectAttributes,
+    ///     actions,
+    ///     "defaultValue");
+    /// </code></example>
+    /// </summary>
+    public BanditResult GetBanditAction(string flagKey,
+                                        string subjectKey,
+                                        IDictionary<string, object?> subjectAttributes,
+                                        IDictionary<string, IDictionary<string, object?>> actions,
+                                        string defaultValue)
+    {
+        return GetBanditDetail(
+            flagKey,
+            ContextAttributes.FromDict(subjectKey, subjectAttributes),
+            actions.ToDictionary(kvp => kvp.Key, kvp => ContextAttributes.FromDict(kvp.Key, kvp.Value)),
+            defaultValue);
+    }
+
+    private BanditResult GetBanditDetail(string flagKey,
+                                        ContextAttributes subject,
+                                        IDictionary<string, ContextAttributes> actions,
+                                        string defaultValue)
+    {
+        InputValidator.ValidateNotBlank(flagKey, "Invalid argument: flagKey cannot be blank");
+
+        // If no actions are given - a valid use case - return the `defaultValue`.
+        bool isBanditFlag = _configurationStore.GetBanditFlags().IsBanditFlag(flagKey);
+
+        if (actions.Count == 0 && isBanditFlag)
+        {
+            // If not graceful mode, raise an exception here?
+            return new(defaultValue);
+        }
+
+
+        // Get the user's flag assignment for the given key.
+        var variation = GetStringAssignment(
+            flagKey,
+            subject.Key,
+            subject,
+            defaultValue);
+
+        // If not a bandit, return the computed String assignment
+        if (!isBanditFlag)
+        {
+            Logger.Warn($"[Eppo SDK] Flag \"{flagKey}\" does not contain a Bandit");
+            return new(variation);
+        }
+
+        try
+        {
+            if (_configurationStore.TryGetBandit(variation, out Bandit? bandit) && bandit != null)
+            {
+                var result = _banditEvaluator.EvaluateBandit(
+                    flagKey,
+                    subject,
+                    actions,
+                    bandit.ModelData);
+
+                var banditActionLog = new BanditLogEvent(
+                    variation,
+                    result,
+                    bandit,
+                    AppDetails.GetInstance().AsDict());
+                _eppoClientConfig.AssignmentLogger.LogBanditAction(banditActionLog);
+                return new BanditResult(variation, result.ActionKey);
+            }
+        }
+        catch (BanditEvaluationException bee)
+        {
+            Logger.Error("[Eppo SDK] Error evaluating bandit, returning variation only: " + bee.Message);
```

Comment:
I guess we don't have the concept of "graceful mode" (or more specifically, the ability to turn it off) in DotNot. Adding this in should be a fast-follow (but doesn't have to be in this PR)